src/main/scala/io/opentargets/etl/backend/drug_beta/Molecule.scala

```diff
@@ -110,11 +103,12 @@ object Molecule extends LazyLogging {
     val full = tradeName
       .join(synonym, Seq("id"), "fullouter")
 
+    // make sure that the arrays aren't left null and are sorted. 
     groupings.foldLeft(full)(
       (df, colName) =>
         df.withColumn(colName,
-          when(col(colName).isNull, array().cast("array<string>"))
-            .otherwise(array_sort(col(colName)))))
+                      when(col(colName).isNull, array().cast("array<string>"))
```

Comment:
@JarrodBaker again `coalesce` is the function you might want to use here. I guess someone from the Spark team realised too much usage of when for nullity check. In this specific case something more like
```scala
coalesce(array_sort(col(colName)), array())
``` 
will work in this case.

Reply:
Thanks for the tip!